### PR TITLE
Sort indices before gathering

### PR DESCRIPTION
**Pull request by zhuofan1123**
In continuous/chunked host mode, sorting indices before gathering can improve bandwidth by enhancing memory locality.

### CODE REVIEW

File: cpp/src/wholememory_ops/functions/gather_func.cu
Comment (linhu-nv) [inline cpp/src/wholememory_ops/functions/gather_func.cu:1]: maybe it's better to set raw_indices the same type as indices

File: cpp/src/wholememory_ops/gather_op.cpp
Comment (linhu-nv) [inline cpp/src/wholememory_ops/gather_op.cpp:1]: In my opinion, whether we use gather_with_sorted_ids should also be related to embedding data type, not just to embedding dim.

**Review by linhu-nv — APPROVED** [review]
Seems good to me.

**Review by BradReesWork — APPROVED** [review]

**Pull request merged**